localflavor/br/validators.py

```diff
@@ -0,0 +1,48 @@
+from django.core.validators import RegexValidator
+from django.utils.translation import ugettext_lazy as _
+
+
+class BRCellPhoneValidator(RegexValidator):
+    """A validator for Brazilian Cell Phone Numbers."""
+
+    def __init__(self, *args, **kwargs):
+        self.regex = '^[(]?[0-9]{2}[)]?[\s]?([0-9]{1})?[0-9]{4}[-]?[0-9]{4}$'
+        self.message = _('Enter a phone number in the format 0000000000 or (00) 00000-0000')
+        self.code = _('Invalid phone')
+        super(BRCellPhoneValidator, self).__init__(*args, **kwargs)
+
+
+class BRTelephoneValidator(RegexValidator):
+    """A validator for Brazilian Telephone Numbers."""
+
+    def __init__(self, *args, **kwargs):
+        self.regex = '^[(]?[0-9]{2}[)]?[\s]?[0-9]{4}[-]?[0-9]{4}$'
+        self.message = _('Enter a telephone in the format 0000000000 or (00) 0000-0000')
+        self.code = 'Invalid telephone'
+        super(BRTelephoneValidator, self).__init__(*args, **kwargs)
+
+
+class BRZipCodeValidator(RegexValidator):
+    """A validator for Brazilian Zip Codes (CEP)."""
+
+    def __init__(self, *args, **kwargs):
+        self.regex = '^[0-9]{5}[-]?[0-9]{3}$'
+        self.message = _('Enter a zip code in the format 00000000 or 00000-000.')
+        self.code = _('Invalid Zip Code')
+        super(BRZipCodeValidator, self).__init__(*args, **kwargs)
+
+
+class BRCNPJValidator(RegexValidator):
```

Comment:
Rather than using a simple reqex validator, the validation logic from the `BRCNPJField` form field needs to be extracted and put here. You'd then use this validator in both the form and model field. This way we won't be including two validation algorithms for the same data type.


The same applies to the `BRCPFValidator` and the `BRCPFField` form field (below).